.circleci/config.yml

```diff
@@ -0,0 +1,55 @@
+version: 2
+
+build_task: &build_task
+  name: Build
+  command: ./gradlew clean assemble check --stacktrace
+verify_task: &verify_task
+  name: Verify Google Java Format
+  command: ./gradlew verGJF
+
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle" }}
```

Comment:
@tylerbenson or anybody else could you please have a look at cache configuration? data dog also adds branch and revision to the key https://github.com/DataDog/dd-trace-java/blob/master/.circleci/config.yml#L45

Reply:
Circleci caches are kind of odd... for any given key, they are immutable, so if you change dependencies mid release (between changes to build.gradle), then those changes won't be cached.  You can get around this by adding additional attributes to make it unique.

They also have a hierarchical cache system, where you can specify multiple key lookups and take the newest most specific match. This means when working from a new branch, it will usually have a new cache to work from. https://circleci.com/docs/2.0/caching/#source-caching